src/SecretManager/SecretManager.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace App\SecretManager;
+
+use Google\Cloud\SecretManager\V1\SecretManagerServiceClient;
+use Google_Client as GoogleClient;
+
+class SecretManager
+{
+    private const SECRET_NAME_CREDENTIALS = 'credentials';
+    private const SECRET_NAME_GA_AUTH_JSON = 'gaAuthJson';
+    private const DEV_PROJECT_ID = 'pmi-hpo-dev';
+
+    private $projectId;
+    private $secretManagerClient;
+
+    public function __construct($useDefaultCredentials)
+    {
+        if (!$useDefaultCredentials) {
+            $basePath = realpath(__DIR__ . '/../../');
+            $keyFile = $basePath . '/dev_config/rdr_key.json';
+            if (!file_exists($keyFile)) {
+                throw new \Exception("Couldn't find $keyFile");
+            }
+            putenv('GOOGLE_APPLICATION_CREDENTIALS=' . $keyFile);
+
+            $googleClient = new GoogleClient();
+            $googleClient->useApplicationDefaultCredentials();
```

Comment:
I don't think you need this. The secret manager client is part of the Google Cloud API library and not related to this class.

Also, what do you think about using the same approach as the `GcsBucketService` class where it looks at the environment to make sure it's local before checking for file in dev_config? And you can specify the key path using the `keyFilePath` config option.